### PR TITLE
test: iframe content and navigation

### PR DESCRIPTION
**Pull request by Rendxn**

### CODE REVIEW

**Comment by Rendxn** [comment]
Throwing `SecurityError: Blocked a frame with origin "https://www.w3schools.com/" from accessing a cross-origin frame.`.

Locally it runs. Should I disable `chromeWebSecurity`?
image

File: cypress/page/iframe.page.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  private iframeTitle: string;
+  private iframeSelector: string;
+  private iframeURL: string;
+
+  constructor() {
+    this.cssNavElement = "a[title='CSS Tutorial']";
+    this.pageURL = "https://www.w3schools.com/html/html_iframe.asp";
+    this.iframeTitle = "#main h1";
+    this.iframeSelector = "iframe[title='W3Schools HTML Tutorial']";
+    this.iframeURL = "https://www.w3schools.com/css/";
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts:13]: can you include something in the name to indicate that is the CSS page, something like: cssFrameUrl or whichever you want

File: cypress/page/iframe.page.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  public goToCssPageInFrame(): void {
+    cy.iframe(this.iframeSelector).find(this.cssNavElement).click();
+  }
+
+  public frameNavigated(): void {
+    cy.frameLoaded(this.iframeSelector, {
+      url: this.iframeURL,
+    });
+  }
```
Comment (holgiosalos) [inline cypress/page/iframe.page.ts:32]: I would make this method private and call it within the goToCssPageInFrame method, or remove the method and paste the code of this method within the goToCssPageInFrame

**Comment by Rendxn** [comment]
Does it make sense? The iframe is same origin. Is it because of the way Cypress works?

**Review by holgiosalos — APPROVED** [review]

**Comment by Rendxn** [comment]
> Does it make sense? The iframe is same origin. Is it because of the way Cypress works?

Tried to recreate it locally and it's inconsistent, so I will leave it like that or add `chromeWebSecurity: false` in the future.

**Pull request merged**